companion/gateway/database.py

```
@@ -0,0 +1,535 @@
+            hourly_activity = {str(i): 0.0 for i in range(24)}
+        
+        # For now, return just hourly activity - weekly would be similar
+        return {'hourly': hourly_activity}
+
+    async def get_active_users_for_reflection(self, days: int = 7) -> List[str]:
+        """Get user IDs of users active in the last N days."""
+        query = f"""
+        SELECT DISTINCT user_id FROM interactions
+        WHERE timestamp > NOW() - INTERVAL '{days} days'
+        ORDER BY timestamp DESC
+        """
```

Ruff (0.13.2)

Possible SQL injection vector through string-based query construction

(S608)